### ISSUE TITLE
[BUG] InvalidCastException & NRE on UsingItemEventArgs

### ISSUE THREAD

**Issue by SebasCapo** [Exiled-Team/EXILED#934]
**Describe the bug**
Simply getting a few exceptions during UsingItemEventArgs, I remember we thought this was people using an item that shouldn't be used, but doesn't seem like it, as this one is a CustomItem of `ItemType.SCP268`:

```cs
public class HatTest : CustomItem
    {

        public override string Name { get; set; } = "TESTITEM";
        public override string Description { get; set; }
            = "TestDescription";
        public override uint Id { get; set; } = 28;
        public override Exiled.API.Features.Spawn.SpawnProperties SpawnProperties { get; set; }
        public override ItemType Type => ItemType.SCP268;
        public override float Weight { get; set; } = 1.15f;

        protected override void SubscribeEvents()
        {
            base.SubscribeEvents();
            Exiled.Events.Handlers.Player.UsingItem += OnItemUsed;
        }

        // This method is called OnItemUsed because I noticed this behavior happening on UsedItemEventArgs, thought it would be worth pointing out!
        private void OnItemUsed(UsingItemEventArgs ev)
        {
            if(Check(ev.Player.CurrentItem))
            {
                Log.Info(ev.Item == null);
                Log.Info(ev.Item.Base == null);
                Log.Info(ev.Item.Serial == null); // unnecessary but will show the NRE

                ev.IsAllowed = false;
                ev.Player.EnableEffect<Amnesia>();
            }
        }

        protected override void UnsubscribeEvents()
        {
            base.UnsubscribeEvents();
            Exiled.Events.Handlers.Player.UsingItem -= OnItemUsed;
        }
```

**To Reproduce**
1. Create a CustomItem, override Type so it's `ItemType.SCP268` or simply copy the code above.
2. Get yourself that custom item.
3. If it tries accessing `UsingItemEventArgs.Item` it will throw either an ICE or NRE.
4. Kiss Joker in the mouth.

**Expected behavior**
The `Item` property shouldn't be null, and be properly cast.

**Server logs**
```wl
[2021-12-09 14:22:15] [ERROR] [Exiled.API] UsedItemEventArgs.ctor: System.InvalidCastException: Specified cast is not valid.
                        at Exiled.Events.EventArgs.UsedItemEventArgs..ctor (Exiled.API.Features.Player player, InventorySystem.Items.Usables.UsableItem item) [0x0001d] in <1976f973928446abb5517f386fbc0323>:0
[2021-12-09 14:22:15] [INFO] [MyPlugin] True
[2021-12-09 14:22:15] [ERROR] [Exiled.API] Method "OnItemUsed" of the class "MyPlugin.Items.HatTest" caused an exception when handling the event "Exiled.Events.Events+CustomEventHandler`1[[Exiled.Events.EventArgs.UsingItemEventArgs, Exiled.Events, Version=4.1.2.0, Culture=neutral, PublicKeyToken=null]]"
[2021-12-09 14:22:15] [ERROR] [Exiled.API] System.NullReferenceException: Object reference not set to an instance of an object
                        at MyPlugin.Items.HatTest.OnItemUsed (Exiled.Events.EventArgs.UsingItemEventArgs ev) [0x0002f] in <f73ea80437524fab900c158732a78f06>:0
                        at Exiled.Events.Extensions.Event.InvokeSafely[T] (Exiled.Events.Events+CustomEventHandler`1[TEventArgs] ev, T arg) [0x00024] in <1976f973928446abb5517f386fbc0323>:0
[2021-12-09 14:22:16] [ERROR] [Exiled.API] UsedItemEventArgs.ctor: System.InvalidCastException: Specified cast is not valid.
                        at Exiled.Events.EventArgs.UsedItemEventArgs..ctor (Exiled.API.Features.Player player, InventorySystem.Items.Usables.UsableItem item) [0x0001d] in <1976f973928446abb5517f386fbc0323>:0
[2021-12-09 14:23:09] [ERROR] [Exiled.API] UsedItemEventArgs.ctor: System.InvalidCastException: Specified cast is not valid.
                        at Exiled.Events.EventArgs.UsedItemEventArgs..ctor (Exiled.API.Features.Player player, InventorySystem.Items.Usables.UsableItem item) [0x0001d] in <1976f973928446abb5517f386fbc0323>:0
[2021-12-09 14:23:10] [ERROR] [Exiled.API] UsedItemEventArgs.ctor: System.InvalidCastException: Specified cast is not valid.
                        at Exiled.Events.EventArgs.UsedItemEventArgs..ctor (Exiled.API.Features.Player player, InventorySystem.Items.Usables.UsableItem item) [0x0001d] in <1976f973928446abb5517f386fbc0323>:0
```

**EXILED Version ("latest" is not a version):**
4.1.2

**Results of `show plugins` command in console:**
Not necessary.

**Additional context**
I believe this one could be tied to #933, but not entirely sure.

**Comment by joker-119** [comment]
The only thing that can cause an ICE is literally if the item they are using is not of the global::UsableItem type, which shouldn't be possible since before this event is called there is a ``if (!(hub.inventory.CurInstance is UsableItem usableItem))`` check (base-game), and the output variable of that cast is what's used for the ctor of the eventargs. 
There is also already a null check where ev.Item can be null before the cast is attempted, soooo...

**Comment by SebasCapo** [comment]
Huh, I see from your PR that this issue is CustomItems specific, I initially thought this was something weird happening on the UsingItem events. 👍 

Either way, I'll wait for 4.1.3 before closing this issue just to make sure it's working as intended. ^^

**Comment by SebasCapo** [comment]
Just tested and seems like this issue was fixed in 4.1.3, thank you! (Tested in 4.1.6 but assuming 4.1.3 is safe too)
